### PR TITLE
Add ESM samples troubleshooting guide

### PR DESCRIPTION
**Pull request by andygup**
This guide outlines various topics for assisting with finding solutions to issues when using modern build tools.

cc @bsvensson

### CODE REVIEW

File: TROUBLESHOOTING.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+**Step 4 – Create a sample.** Always try to provide a simple sample when posting your question to the Esri Community Site. There great online options for sharing samples including [codepen](https://codepen.io/), [github.com](https://github.com/), [codesandbox](https://codesandbox.io/) or [StackBlitz](https://stackblitz.com/) are great for providing reproducible examples. Code snippets often do not provide enough information about a project because they don’t include information on configuration, dependencies or coding patterns. Providing a working sample is essential to allow others to quickly reproduce your issue. Sometimes the process of creating the sample helps solve the problem.
+
+## Understand the cause of an issue
+
+With all the various parts in modern build systems, determining where the problem is occurring is critical. There are typically three categories where problems occur: configuration, bundler/framework, or the Maps SDK.
```
Comment (lboyd93) [inline TROUBLESHOOTING.md:26]:
```suggestion
With all the various parts in modern build systems, determining where the problem is occurring is critical. There are typically three categories where problems occur: configuration, bundler/framework, or the JS Maps SDK.
```

File: TROUBLESHOOTING.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+**Problem 1:** `404` errors that indicate a file is missing in a production build, but dev build works fine.
+
+_Potential Resolution:_ Check the bundler or framework documentation on setting the base path for production builds. See [example](https://github.com/Esri/jsapi-resources/issues/436).  
+
+**Problem 2:** Module parse failed: Unexpected token. You may need an appropriate loader to handle this file type. 
+
+_Potential Resolution:_ These are typically related to transpiling. You may be missing a needed transpiler plugin or need to upgrade to a newer version of the module bundler. The error messages may point to one or more Maps SDK modules. See [example](https://github.com/Esri/jsapi-resources/issues/424).
+*	Get a screenshot of the full error message
+*	Narrow down an Maps SDK version where the break occurred
```
Comment (lboyd93) [inline TROUBLESHOOTING.md:45]:
```suggestion
*	Narrow down a JS Maps SDK version where the break occurred
```

File: TROUBLESHOOTING.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+### Bundler/Framework issues
+Bundlers and frameworks do occasionally have their own bugs. If you are working with a major framework or bundler, they have very active communities. Searching for the error message using a general internet search is typically the fastest way to get information and hints.
+* Configuration issues
+* Component issues
+* Syntax errors
+* Life-cycle problems
+
+#### Example Bundler/Framework issues
+
+We do come across bugs in module bundlers and frameworks that are unrelated to the Maps SDK. Sometimes these bugs have already been documented. Sometimes, we recommend you open an issue on the framework or bundler repository, that’s where many examples of these issues can be found, for example:
```
Comment (lboyd93) [inline TROUBLESHOOTING.md:60]:
```suggestion
We do come across bugs in module bundlers and frameworks that are unrelated to the JS Maps SDK. Sometimes these bugs have already been documented. Sometimes, we recommend you open an issue on the framework or bundler repository, that’s where many examples of these issues can be found, for example:
```

**Review by lboyd93 — APPROVED** [review]
Great information, @andygup! Added some suggestions.

The only thing I am not sure about is the shorthand for the product. I may be wrong with my suggestions so feel free to ignore them if it should be `Maps SDK`. If not, the last section would need some updating.

**Comment by andygup** [comment]
I think the name change is fine, thanks @lboyd93.

File: TROUBLESHOOTING.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+**Step 4 – Create a sample.** Always try to provide a simple sample when posting your question to the Esri Community Site. There are great online options for sharing samples including [codepen](https://codepen.io/), [github.com](https://github.com/), [codesandbox](https://codesandbox.io/) or [StackBlitz](https://stackblitz.com/). Code snippets often do not provide enough information about a project because they don’t include information on configuration, dependencies or coding patterns. Providing a working sample is essential to allow others to quickly reproduce your issue. Sometimes the process of creating the sample helps solve the problem.
+
+## Understand the cause of an issue
+
+With all the various parts in modern build systems, determining where the problem is occurring is critical. There are typically three categories where problems occur: configuration, bundler/framework, or the JS Maps SDK.
+
+### Configuration issues
```
Comment (dasa) [inline TROUBLESHOOTING.md:28]:
```suggestion
## Configuration issues
```

File: TROUBLESHOOTING.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+_Potential Resolution:_ These are typically related to transpiling. You may be missing a needed transpiler plugin or need to upgrade to a newer version of the module bundler. The error messages may point to one or more Maps SDK modules. See [example](https://github.com/Esri/jsapi-resources/issues/424).
+*	Get a screenshot of the full error message
+*	Narrow down the JS Maps SDK version where the break occurred
+
+**Problem 3:** Error loading a .wasm or .woff file.
+
+_Potential Resolution:_ Configure your web server with the correct MIME types. See [example](https://developers.arcgis.com/javascript/latest/install-and-set-up/#web-server-hosting-configuration).
+
+### Bundler/Framework issues
```
Comment (dasa) [inline TROUBLESHOOTING.md:51]:
```suggestion
## Bundler/Framework issues
```

File: TROUBLESHOOTING.md [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+#### Example Bundler/Framework issues
+
+We do come across bugs in module bundlers and frameworks that are unrelated to the JS Maps SDK. Sometimes these bugs have already been documented. Sometimes, we recommend you open an issue on the framework or bundler repository, that’s where many examples of these issues can be found, for example:
+
+*	React issues: https://github.com/facebook/react/issues 
+*	Angular issues: https://github.com/angular/angular-cli/issues 
+*	Rollup issues: https://github.com/rollup/rollup/issues 
+*	Webpack issues: https://github.com/webpack/webpack/issues 
+
+### Maps SDK issues
```
Comment (dasa) [inline TROUBLESHOOTING.md:67]:
```suggestion
## Maps SDK issues
```

**Review by dasa — CHANGES_REQUESTED** [review]
Update heading levels to match TOC

File: README.md [hunk 7]
```diff
@@ -21,6 +21,10 @@ A collection of resources for developers using the [ArcGIS Maps SDK for JavaScri
 ## Instructions
 Refer to the README files in each subdirectory of this repo for specific instructions on how to use a particular resource.
 
+## Troubleshooting
+For recommendations on identifying and solving issues in the ES module samples, see the [troubleshooting guide](./TROUBLESHOOTING.md).
+
+
```
Comment (dasa) [inline README.md:27]: Delete new line

File: TROUBLESHOOTING.md [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+
+## Understand the cause of an issue
+
+With all the various parts in modern build systems, determining where the problem is occurring is critical. There are typically three categories where problems occur: configuration, bundler/framework, or the JS Maps SDK.
+
+### Configuration issues
+The most common errors are related to project configuration. Most of these types of issues look like bugs, but oftentimes they are not:
+*	Module bundler – e.g. webpack and rollup configuration
+*	Framework – e.g. Angular, React configuration
+*	Package.json – e.g. dependency version mismatches
```
Comment (dasa) [inline TROUBLESHOOTING.md:32]:
```suggestion
*	package.json – e.g. dependency version mismatches
```

It's odd seeing that capitalized.

File: TROUBLESHOOTING.md [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+* Life-cycle problems
+
+#### Example Bundler/Framework issues
+
+We do come across bugs in module bundlers and frameworks that are unrelated to the JS Maps SDK. Sometimes these bugs have already been documented. Sometimes, we recommend you open an issue on the framework or bundler repository, that’s where many examples of these issues can be found, for example:
+
+*	React issues: https://github.com/facebook/react/issues 
+*	Angular issues: https://github.com/angular/angular-cli/issues 
+*	Rollup issues: https://github.com/rollup/rollup/issues 
+*	Webpack issues: https://github.com/webpack/webpack/issues
```
Comment (dasa) [inline TROUBLESHOOTING.md:65]: Sort?

File: TROUBLESHOOTING.md [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+*	Errors that occur in a Maps SDK module
+*	Errors that can be reproduced using the Maps SDK in plain JavaScript without a framework
+
+#### Example Maps SDK issue
+
+**Errors in production builds.** While very rare, sometimes the production build process of a framework creates errors that don’t show up in developer builds. These are often related to bundling, minification or transpiling. Look for hints in the error message that might point to functionality inside a Maps SDK module.
+
+[Example - ReferenceError: Cannot access “e” before initialization](https://github.com/Esri/jsapi-resources/issues/309). In this case, when the error stack dump was expanded in the browser console, you could see file names that appear to be from the Maps SDK such as _renderLegendForLayer_ and _renderLegendForElement_, these are excellent hints.
+
+Verified errors in production builds can be reported on the [Maps SDK Community site](https://community.esri.com/t5/arcgis-javascript-maps-sdk-questions/bd-p/arcgis-api-for-javascript-questions), or if it’s specifically related to an ES modules sample on this repo, then you can open a github issue [here](https://github.com/Esri/jsapi-resources/issues).
```
Comment (dasa) [inline TROUBLESHOOTING.md:79]:
```suggestion
Verified errors in production builds can be reported on the [Maps SDK Community site](https://community.esri.com/t5/arcgis-javascript-maps-sdk-questions/bd-p/arcgis-api-for-javascript-questions), or if it’s specifically related to an ES modules sample on this repo, then you can open a GitHub issue [here](https://github.com/Esri/jsapi-resources/issues).
```

File: TROUBLESHOOTING.md [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+
+**Regressions.** This is when Maps SDK functionality works in one release, and then the same functionality is broken in the next one. It’s important to insure the bugs are related to some other dependency in package.json. When testing for regressions, you will change only the version of @arcgis/core and keep all other dependencies constant. 
+
+Most Maps SDK regressions can be re-created in a plain JavaScript application, without the framework or bundler. The ESM CDN is an easy way to test and prototype these types of issues: https://github.com/Esri/jsapi-resources/tree/master/esm-samples/jsapi-esm-cdn. 
+
+Verified regressions can be reported to[ Esri Technical Support](https://support.esri.com/en/contact-tech-support) or the [Maps SDK Community site](https://community.esri.com/t5/arcgis-javascript-maps-sdk-questions/bd-p/arcgis-api-for-javascript-questions).
+
+**New bugs.** The best practice for verifying these is to re-create them in a simple, plain JavaScript application, without the framework or bundler. The ESM CDN is an easy way to test for these types of issues, here is sample application.
```
Comment (dasa) [inline TROUBLESHOOTING.md:87]: Period after "issues"? Link to sample?

File: TROUBLESHOOTING.md (hunk 9 above)
Comment (andygup) [reply TROUBLESHOOTING.md]: Done.

File: TROUBLESHOOTING.md (hunk 11 above)
Comment (andygup) [reply TROUBLESHOOTING.md]: Fixed.

File: README.md (hunk 7 above)
Comment (andygup) [reply README.md:27]: My local copy looks correct.

**Review by dasa — APPROVED** [review]

File: TROUBLESHOOTING.md [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+
+**Step 4 – Create a sample.** Always try to provide a simple sample when posting your question to the Esri Community Site. There are great online options for sharing samples including [codepen](https://codepen.io/), [github.com](https://github.com/), [codesandbox](https://codesandbox.io/) or [StackBlitz](https://stackblitz.com/). Code snippets often do not provide enough information about a project because they don’t include information on configuration, dependencies or coding patterns. Providing a working sample is essential to allow others to quickly reproduce your issue. Sometimes the process of creating the sample helps solve the problem.
+
+## Understand the cause of an issue
+
+With all the various parts in modern build systems, determining where the problem is occurring is critical. There are typically three categories where problems occur: configuration, bundler/framework, or the JS Maps SDK.
+
+## Configuration issues
```
Comment (odoe) [inline TROUBLESHOOTING.md:28]: Empty newline after header items, couple cases here. Do newlines matter if header 2 or 3?

**Review by odoe — APPROVED** [review]
Nice work! This will be a big help!

File: TROUBLESHOOTING.md (hunk 12 above)
Comment (andygup) [reply TROUBLESHOOTING.md:28]: Ah yes, I can remove them. Github markdown ignores them, but I'll fix them for consistency.

**Pull request merged**